oak/server/rust/oak_runtime/src/node/grpc/server/mod.rs

```diff
@@ -258,6 +267,54 @@ impl Service<http::Request<hyper::Body>> for HttpRequestHandler {
     }
 }
 
+impl From<OakLabelError> for tonic::Status {
+    fn from(v: OakLabelError) -> Self {
+        match v {
+            OakLabelError::MissingLabel => tonic::Status::invalid_argument("Missing Oak Label"),
+            OakLabelError::MultipleLabels => tonic::Status::invalid_argument("Multiple Oak Labels"),
+            OakLabelError::InvalidLabel => tonic::Status::invalid_argument("Invalid Oak Label"),
+        }
+    }
+}
+
+enum OakLabelError {
+    MissingLabel,
+    MultipleLabels,
+    InvalidLabel,
+}
+
+/// Returns the [`Label`] defined as part of the the metadata of an incoming gRPC request.
+///
+/// Returns an error if there is not exactly one label specified by the caller.
+fn get_oak_label(metadata_map: &MetadataMap) -> Result<Label, OakLabelError> {
+    let labels = metadata_map
+        .get_all_bin(OAK_LABEL_GRPC_METADATA_KEY)
+        .iter()
+        .collect::<Vec<_>>();
+    if labels.is_empty() {
+        warn!(
+            "incorrect number of gRPC labels found: {}, expected: 1",
```

Comment:
Why couldn't there be more than 1 label?
Maybe add a comment

Reply:
Done.